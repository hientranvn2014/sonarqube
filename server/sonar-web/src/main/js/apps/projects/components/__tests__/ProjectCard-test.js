/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import ProjectCard from '../ProjectCard';

const PROJECT = {
  analysisDate: '2017-01-01',
  leakPeriodDate: '2016-12-01',
  key: 'foo',
  name: 'Foo',
  tags: []
};
const MEASURES = {
  alert_status: 'OK',
  reliability_rating: '1.0',
  sqale_rating: '1.0',
  new_bugs: 12
};

jest.mock('moment', () => () => ({
  format: () => 'March 1, 2017 9:36 AM',
  fromNow: () => 'a month ago'
}));

describe('overall status project card', () => {
  it('should never display analysis date', () => {
    expect(
      shallow(<ProjectCard measures={{}} project={PROJECT} />).find('.project-card-dates').exists()
    ).toBeFalsy();
  });

  it('should display loading', () => {
    const measures = { ...MEASURES, sqale_rating: undefined };
    expect(
      shallow(<ProjectCard project={PROJECT} />)
        .find('.boxed-group')
        .hasClass('boxed-group-loading')
    ).toBeTruthy();
    expect(
      shallow(<ProjectCard measures={measures} project={PROJECT} />)
        .find('.boxed-group')
        .hasClass('boxed-group-loading')
    ).toBeTruthy();
  });

  it('should not display the quality gate', () => {
    const project = { ...PROJECT, analysisDate: undefined };
    expect(
      shallow(<ProjectCard measures={MEASURES} project={project} />)
        .find('ProjectCardQualityGate')
        .exists()
    ).toBeFalsy();
  });

  it('should display tags', () => {
    const project = { ...PROJECT, tags: ['foo', 'bar'] };
    expect(shallow(<ProjectCard project={project} />).find('TagsList').exists()).toBeTruthy();
  });

  it('should private badge', () => {
    const project = { ...PROJECT, visibility: 'private' };
    expect(
      shallow(<ProjectCard type="overall" project={project} />).find('PrivateBadge').exists()
    ).toBeTruthy();
  });

  it('should display the overall measures and quality gate', () => {
    expect(shallow(<ProjectCard measures={MEASURES} project={PROJECT} />)).toMatchSnapshot();
  });
});

describe('leak project card', () => {
  it('should display analysis date and leak start date', () => {
    const card = shallow(<ProjectCard type="leak" measures={MEASURES} project={PROJECT} />);
    expect(card.find('.project-card-dates').exists()).toBeTruthy();
    expect(card.find('.project-card-dates').find('span').getNodes()).toHaveLength(2);
  });

  it('should not display analysis date or leak start date', () => {
    const project = { ...PROJECT, analysisDate: undefined };
    const card = shallow(<ProjectCard type="leak" measures={MEASURES} project={project} />);
    expect(card.find('.project-card-dates').exists()).toBeFalsy();
  });

  it('should display loading', () => {
    const measures = { ...MEASURES, new_bugs: undefined };
    expect(
      shallow(<ProjectCard type="leak" measures={measures} project={PROJECT} />)
        .find('.boxed-group')
        .hasClass('boxed-group-loading')
    ).toBeTruthy();
  });

  it('should display tags', () => {
    const project = { ...PROJECT, tags: ['foo', 'bar'] };
    expect(
      shallow(<ProjectCard type="leak" project={project} />).find('TagsList').exists()
    ).toBeTruthy();
  });

  it('should private badge', () => {
    const project = { ...PROJECT, visibility: 'private' };
    expect(
      shallow(<ProjectCard type="leak" project={project} />).find('PrivateBadge').exists()
    ).toBeTruthy();
  });

  it('should display the leak measures and quality gate', () => {
    expect(
      shallow(<ProjectCard type="leak" measures={MEASURES} project={PROJECT} />)
    ).toMatchSnapshot();
  });
});
