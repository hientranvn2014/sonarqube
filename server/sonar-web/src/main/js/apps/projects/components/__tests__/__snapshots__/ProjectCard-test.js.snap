// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display loading 1`] = `
<div
  className="boxed-group project-card boxed-group-loading"
  data-key="foo"
>
  <div
    className="boxed-group-actions text-right"
  />
  <div
    className="boxed-group-header"
  >
    <h2
      className="project-card-name"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/dashboard",
            "query": Object {
              "id": "foo",
            },
          }
        }
      >
        Foo
      </Link>
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <ProjectCardMeasures />
  </div>
</div>
`;

exports[`should display tags 1`] = `
<div
  className="boxed-group project-card boxed-group-loading"
  data-key="foo"
>
  <div
    className="boxed-group-actions text-right"
  >
    <TagsList
      allowUpdate={false}
      customClass="spacer-left"
      tags={
        Array [
          "foo",
          "bar",
        ]
      }
    />
  </div>
  <div
    className="boxed-group-header"
  >
    <h2
      className="project-card-name"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/dashboard",
            "query": Object {
              "id": "foo",
            },
          }
        }
      >
        Foo
      </Link>
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <ProjectCardMeasures />
  </div>
</div>
`;

exports[`should not display analysis date 1`] = `undefined`;
